app/include/lwip/app/dhcpserver.h

```diff
@@ -19,7 +21,9 @@ typedef struct dhcps_msg {
         uint8_t chaddr[16];
         uint8_t sname[64];
         uint8_t file[128];
-        uint8_t options[312];
+	// Recommendation from Espressif:
+	// To avoid crash in DHCP big packages modify option length from 312 to MTU - IPHEAD(20) - UDPHEAD(8) - DHCPHEAD(236).
+        uint8_t options[IP_FRAG_MAX_MTU - 20 - 8 - 236];

```

Comment:
> It also isn't clear that IP_FRAG_MAX_MTU is the upper limit

I've chosen `IP_FRAG_MAX_MTU` to implement Espressif's guidance for MTU since our lwip is having `IP_FRAG_USES_STATIC_BUF` enabled in [lwipopts.h](/nodemcu/nodemcu-firmware/blob/dev/app/include/lwipopts.h#L555). It seemed to be the cheapest solution.